Lua/Lua.sublime-syntax

```diff
@@ -8,18 +8,16 @@ file_extensions:
 scope: source.lua
 contexts:
   main:
-    - match: '\b(function)\b\s*([a-zA-Z_]\w*)?\s*(([.]\s*[a-zA-Z_]\w*\s*)*)\s*(([.:])?\s*[a-zA-Z_]\w*)?\s*(\()([^)]*)(\))?'
+    - match: '\b(function)\b\s*(([a-zA-Z_]\w*\s*[.]\s*)*)?\s*([a-zA-Z_]\w*\s*[:])?\s*([a-zA-Z_]\w*)?\s*(\()([^)]*)(\))?'
```

Comment:
I think this regex would benefit from being split up and using variables (for the identifiers). It's not really readable at the moment.